Source/WebCore/style/PseudoElementRequest.h

```diff
@@ -33,23 +34,23 @@ namespace WebCore::Style {
 class PseudoElementRequest {
 public:
     PseudoElementRequest(PseudoId pseudoId, std::optional<StyleScrollbarState> scrollbarState = std::nullopt)
-        : pseudoId(pseudoId)
-        , scrollbarState(scrollbarState)
+        : m_identifier(pseudoId)
+        , m_scrollbarState(scrollbarState)
     {
     }
 
     PseudoElementRequest(PseudoId pseudoId, const AtomString& pseudoElementNameArgument)
-        : pseudoId(pseudoId)
-        , pseudoElementNameArgument(pseudoElementNameArgument)
+        : m_identifier({ pseudoId, pseudoElementNameArgument })
     {
-        ASSERT(pseudoId == PseudoId::Highlight || pseudoId == PseudoId::ViewTransitionGroup || pseudoId == PseudoId::ViewTransitionImagePair || pseudoId == PseudoId::ViewTransitionOld || pseudoId == PseudoId::ViewTransitionNew);
     }
 
-    PseudoId pseudoId;
-    std::optional<StyleScrollbarState> scrollbarState;
+    PseudoId pseudoId() const { return m_identifier.pseudoId; }
+    const AtomString& pseudoElementNameArgument() const { return m_identifier.pseudoElementNameArgument; }
+    const std::optional<StyleScrollbarState>& scrollbarState() const { return m_scrollbarState; }
 
-    // highlight name for ::highlight or view transition name for view transition pseudo elements.
-    AtomString pseudoElementNameArgument;
+private:
+    PseudoElementIdentifier m_identifier;
```

Comment:
Should this only be an implementation detail? Maybe we want to expose it as a constructor argument or expose a getter?

Reply:
Yeah that seems like a good idea.